### PR TITLE
Require PIN for registering an  hotp credential

### PR DESCRIPTION
**Pull request by sosthene-nitrokey**
Motivation: https://github.com/Nitrokey/nitrokey-hotp-verification/issues/30

### CODE REVIEW

File: src/credential.rs [hunk 1]
```diff
@@ -213,6 +213,12 @@ impl CredentialFlat {
 
     /// Update credential fields with new values, and save
     pub fn update_from(&mut self, update_req: UpdateCredential) -> Result<(), Status> {
+        // Updating ReverseHOTP is disabled without PIN
```
Comment (robin-nitrokey) [inline src/credential.rs:216]:
```suggestion
        // Updating ReverseHOTP is disabled
```

AFAIS you could create a ReverseHotp credential with PIN protection but would still not be able to update it.

**Comment by tlaurion** [comment]
@sosthene-nitrokey can you link to parallel work for hotp-verification and Heads code needed to deal with this?

**Comment by stv0g** [comment]
My testing show registering a TOTP secret still requires no PIN to be set.
Is this indented? WHy is there this asymmetry between HOTP and TOTP?

**Comment by sosthene-nitrokey** [comment]
The use case is different and leads to different threat models.

HOTP is used so that the key can verify that the code sent to it is correct. Being able to create an HOTP credential could allow verification against a newly created credential instead of the real one.

On the other hand TOTP is used to authenticate the key to a service. Being able to create false credential would only mean that the key would not be able to be authenticated, leading to an explicit failure, instead of authentication working when it shouldn't.

You should not that it it also not possible to create a pin-protected without first presenting the PIN.

**Comment by robin-nitrokey** [comment]
To be precise, HOTP and TOTP credentials are handled identically.  What is different and affected by this PR is *Reverse* HOTP, sometimes shortened to HOTP.

**Pull request merged**